File: 백준/Silver/10828. 스택/스택.py
import sys 
num = int(sys.stdin.readline())

def solution(num):
    stack = []
    top = -1
    for i in range(num):
        line = sys.stdin.readline()
        inputs = line.split()
        if inputs[0] == "pop":
            if len(stack) > 0:
                print(stack.pop())
            else:
                print(-1)
        elif inputs[0] == "top":
            if len(stack) >= 1:
                element = stack.pop()
                print(element)
                stack.append(element)
            elif len(stack) == 0:
                print(-1)
        elif inputs[0] == "size":
            print(len(stack))
        elif inputs[0] == "empty":
            if stack:
                  print(0)
            else:
                  print(1)
        elif inputs[0] == "push":
            num = int(inputs[1])
            stack.append(num)  

solution(num)  

File: 백준/Silver/10828. 스택/스택.swift
struct Stack {
    var arr: [Int] = []
    var size: Int {
        return arr.count
    }
    var empty: Int {
        return arr.isEmpty ? 1 : 0
    }
    var top: Int {
        return arr.isEmpty ? -1 : arr[arr.count - 1]
    }
    mutating func push(_ num: Int){
        self.arr.append(num)
    }
    mutating func pop() -> Int {
        return self.arr.popLast() ?? -1
    }
}

var stack = Stack()
let T = Int(readLine()!)!
for _ in 0..<T{
    let input = readLine()!.split(separator: " " )
    if input.count >= 2{
        stack.push( Int(input[1])!)
    }
    else{
        // pop, size, empty, top
        switch input[0]{
            case "pop":
                print(stack.pop())
            case "size":
                print(stack.size)
            case "empty":
                print(stack.empty)
            case "top":
                print(stack.top)
            default:
                break
        }
    }
}